extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -84,23 +86,15 @@ func (a *genericActuator) Reconcile(ctx context.Context, worker *extensionsv1alp
 
 	// During the time a rolling update happens we do not want the cluster autoscaler to interfer, hence it
 	// is removed for now.
-	var (
-		clusterAutoscalerUsed = extensionsv1alpha1helper.ClusterAutoscalerRequired(worker.Spec.Pools)
-		rollingUpdate         = false
-	)
+	var clusterAutoscalerUsed = extensionsv1alpha1helper.ClusterAutoscalerRequired(worker.Spec.Pools)
+	rollingUpdateInProgress := a.isRollingUpdateInProgress(wantedMachineDeployments, existingMachineClassNames)
```

Comment:
```suggestion
	var (
		clusterAutoscalerUsed = extensionsv1alpha1helper.ClusterAutoscalerRequired(worker.Spec.Pools)
		rollingUpdateInProgress = a.isRollingUpdateInProgress(wantedMachineDeployments, existingMachineClassNames
	)
```
Minor suggestion keeping uniformity in mind.

Reply:
> This changes the logic that waits for the rolling update to finish.
Currently when triggering a rolling update (e.g volume resize) with maxUnavailability = 0, it does not wait for the new machine to join the cluster. This is a bug in my opinion (see Release notes).

In certain cases it did not wait for the Rolling update to complete. So the RollingUpdate condition was removed too early.